Initial update of pi-gadget for Ubuntu Core 26

nicoharnois opened the pull request:
- Added core26 build for arm64
- Align kernel command line and piboot config with the server branch

nicoharnois committed:
snapcraft: add core26 build for arm64

Signed-off-by: Nicolas Harnois <nicolas.harnois@canonical.com>

nicoharnois committed:
Align kernel command line and piboot config with the server gadget

multipath=off is not added for core as core is not shipped with
multipathd daemon.

Signed-off-by: Nicolas Harnois <nicolas.harnois@canonical.com>

alfonsosanchezbeato commented on configs/cmdline.txt:
```diff
@@ -1 +1 @@
-console=serial0,115200 dwc_otg.lpm_enable=0 console=tty1 rng_core.default_quality=700 vt.handoff=2 quiet splash
+console=serial0,115200 dwc_otg.lpm_enable=0 console=tty1 panic=10 rng_core.default_quality=700 vt.handoff=2 quiet splash
```
We use `panic=-1` in our default grub configuration, so we revert faster if there was a kernel issue.

nicoharnois replied:
OK, I will update this

alfonsosanchezbeato commented on configs/config.txt:
```diff
@@ -25,19 +25,22 @@ disable_overscan=1
 
+# Enable the KMS ("Full" KMS) graphics overlay, and allocate 128Mb to the
+# GPU memory
+dtoverlay=vc4-kms-v3d,cma-128
```
Is this working ok with the splash screen? Or any other graphics program fwiw.

nicoharnois replied:
Actually, it is not. Thanks for the heads up.

I tried to unpack the snap, update all the boot-assets from https://github.com/raspberrypi/firmware/raw/master/boot/, and repack the snap, but it does not seem to be enough.

Still, the patch to increase the overlay file path should be included: https://github.com/raspberrypi/firmware/commit/55c0555f30b4ff1ad20a3dd4db535b0d013c0a40

nicoharnois replied:
On RPi 5, the slash screen is properly displayed.
After reflashing the FW with the fix from https://github.com/raspberrypi/firmware/issues/2004, I am also able to start ubuntu-frame.

alfonsosanchezbeato commented on configs/config.txt:
```diff
@@ -49,6 +52,13 @@ dtoverlay=vc4-fkms-v3d,cma-128
-arm_64bit=1
-dtoverlay=dwc2
+[pi4]
+max_framebuffers=2
```
I think that this one is legacy and could be removed

nicoharnois replied:
You mean the max_framebuffers=2 parameter for pi4? 
We still have it on the classic branch, for resolute:

https://github.com/canonical/pi-gadget/blob/classic/configs/resolute-arm64-server/config.txt#L47

Please confirm if I should remove it.

alfonsosanchezbeato replied:
But according to this, RPi4 sets it to 2 by default: https://www.raspberrypi.com/documentation/computers/legacy_config_txt.html , and here we are setting it specifically to 2. Maybe worth asking @waveform80 if worth keeping, here and in classic.

nicoharnois replied:
Removed in the latest commit after Dave's confirmation.

alfonsosanchezbeato reviewed (commented):
Thank you, some comments

nicoharnois committed:
Let's use panic=-1 so we revert faster if there was a kernel issue.

Signed-off-by: Nicolas Harnois <nicolas.harnois@canonical.com>

nicoharnois committed:
Remove useless parameter setting for pi4

max_framebuffers is set by default to 2 based on the RPi documentation.
Let's remove this explicit setting.

Signed-off-by: Nicolas Harnois <nicolas.harnois@canonical.com>

alfonsosanchezbeato reviewed (approved):
Thank you!
